### PR TITLE
feat: add "parquet_scan" function

### LINKED ISSUE #1098: Read from http endpoints
## Summary

E.g. `select * from read_url('https://huggingface.co/...')`

See https://huggingface.co/blog/hub-duckdb

### PR DESCRIPTION
**Pull request by universalmind303**
closes #1098 

note: this is currently limited to `parquet` for a single file. 

Todo: 
- add support for an array of files `parquet_scan([...])`

### CODE REVIEW

File: crates/sqlbuiltins/src/functions.rs [hunk 1]
```diff
@@ -466,6 +469,50 @@ impl TableFunc for ReadSnowflake {
...
+                let accessor = HttpAccessor::try_new(url)
+                    .await
+                    .map_err(|e| BuiltinError::Access(Box::new(e)))?;
+                let prov = ParquetTableProvider::from_table_accessor(accessor, false)
+                    .await
+                    .unwrap();
+
+                Ok(Arc::new(prov))
+            }
+            _ => Err(BuiltinError::Unimplemented("invalid number of arguments")),
```
Comment (scsmithr) [inline crates/sqlbuiltins/src/functions.rs:511]:
```suggestion
            _ => Err(BuiltinError::InvalidNumArgs),
```

File: crates/sqlbuiltins/src/functions.rs [hunk 2]
```diff
@@ -466,6 +469,50 @@ impl TableFunc for ReadSnowflake {
...
+            // parquet_scan(url)
+            1 => {
+                let mut args = args.into_iter();
+                let url = string_from_scalar(args.next().unwrap())?;
+                let accessor = HttpAccessor::try_new(url)
+                    .await
+                    .map_err(|e| BuiltinError::Access(Box::new(e)))?;
+                let prov = ParquetTableProvider::from_table_accessor(accessor, false)
+                    .await
+                    .unwrap();
```
Comment (scsmithr) [inline crates/sqlbuiltins/src/functions.rs:507]:
```suggestion
                let prov = ParquetTableProvider::from_table_accessor(accessor, false)
                    .await?;
```

File: crates/sqlbuiltins/src/functions.rs (hunk 1 above)
Comment (scsmithr) [reply crates/sqlbuiltins/src/functions.rs:511]: Or maybe

```
.map_err(|e| BuiltinError::Access(Box::new(e)))?;
```

**Review by scsmithr — COMMENTED** [review]
Just some minor stuff, but otherwise this looks good to me.

We can add SLTs as a follow up (I'd be fine with just using a huggingface url to assert that it works).

File: crates/sqlbuiltins/src/functions.rs (hunk 2 above)
Comment (universalmind303) [reply crates/sqlbuiltins/src/functions.rs]: good catch, fixed!

**Pull request merged**